### PR TITLE
EOS-22277: Accommodating openldap mini prov with s3 prov

### PR DESCRIPTION
**Pull request by shalakadharap**

### CODE REVIEW

File: scripts/provisioning/configcmd.py [hunk 1]
```diff
@@ -99,6 +101,42 @@ def process(self, configure_only_openldap = False, configure_only_haproxy = Fals
...
+    handler = SimpleProcess(cmd)
+    stdout, stderr, retcode = handler.run()
+    self.logger.info(f'output of setup_ldap.sh: {stdout}')
+    if retcode != 0:
+      self.logger.error(f'error of setup_ldap.sh: {stderr}')
+      raise S3PROVError(f"{cmd} failed with err: {stderr}, out: {stdout}, ret: {retcode}")
+    else:
+      self.logger.warning(f'warning of setup_ldap.sh: {stderr}')
+
+    if os.path.isfile("/opt/seagate/cortx/s3/install/ldap/rsyslog.d/slapdlog.conf"):
```
Comment (bkirunge7) [inline scripts/provisioning/configcmd.py:121]: Openldap logging configuration need to be moved to openldap mini provisioning

Comment (shalakadharap) [reply scripts/provisioning/configcmd.py:121]: it is present in openldap mini prov. I forgot to remove it from this place

**Review by bkirunge7 — APPROVED** [review]

**Comment by shahidSTX** [comment]
Please remove the commented out functions, rest all looks good to me.

**Review by shahidSTX — APPROVED** [review]
LGTM

File: scripts/provisioning/setupcmd.py [hunk 2]
```diff
@@ -594,24 +594,22 @@ def get_config_param_for_BG_delete_account(self):
...
-    self.deleteattribute(conn, dn, "olcMirrorMode")
-
   def deleteattribute(self, conn, dn, attr_to_delete):
     mod_attrs = [(ldap.MOD_DELETE, attr_to_delete, None)]
     try:
       conn.modify_s(dn, mod_attrs)
     except:
       self.logger.info('Attribute '+ attr_to_delete + ' is not configured for dn '+ dn)
+
+  def modify_attribute(self, dn, attribute, value):
```
Comment (vimalraghwani20) [inline scripts/provisioning/setupcmd.py:604]: why can't we directly import from py-utils ?
we can avoid re-writing the same function

Comment (shalakadharap) [reply scripts/provisioning/setupcmd.py:604]: plz suggest how to do it.

Comment (shalakadharap) [reply scripts/provisioning/setupcmd.py]: Fixed

File: scripts/provisioning/setupcmd.py [hunk 3]
```diff
@@ -594,24 +584,43 @@ def get_config_param_for_BG_delete_account(self):
...
+            ldap_conn.modify_s(dn, mod_attrs)
+        except:
+            self.logger.error('Error while modifying attribute- '+ attribute )
+            raise Exception('Error while modifying attribute' + attribute)
+        ldap_conn.unbind_s()
+
+  def search_and_delete_attribute(self, dn, attr_to_delete):
+        conn = ldap.initialize("ldapi://")
+        conn.sasl_non_interactive_bind_s('EXTERNAL')
+        ldap_result_id = conn.search_s(dn, ldap.SCOPE_BASE, None, [attr_to_delete])
```
Comment (bkirunge7) [inline scripts/provisioning/setupcmd.py:603]: Separate this in different function getCountofRecords()

Comment (shalakadharap) [reply scripts/provisioning/setupcmd.py:603]: Fixed

File: scripts/provisioning/setupcmd.py (hunk 2 above)
Comment (shalakadharap) [reply scripts/provisioning/setupcmd.py:612]: Tried but its failing with nested used packages dependency

**Pull request merged**